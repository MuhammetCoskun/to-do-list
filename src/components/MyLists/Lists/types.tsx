import { ToDoListItemType } from "../../ShowList/ToDoList/types";

export type ListType = {
  id: string;
  title: string;
  todos: ToDoListItemType[];
};

export type ListsProps = {
  lists: ListType[];
  onChooseList: (arg: string) => void;
};
