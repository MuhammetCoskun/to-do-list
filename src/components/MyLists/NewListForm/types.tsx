export type NewListFormProps = {
  onAddList: (listName: string) => void;
};
