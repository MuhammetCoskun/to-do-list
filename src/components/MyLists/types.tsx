import { ListType } from "./Lists/types";

export type MyListsProps = {
  onChooseList: (arg: string) => void;
  onAddList: (title: string) => void;
  lists: ListType[];
};
