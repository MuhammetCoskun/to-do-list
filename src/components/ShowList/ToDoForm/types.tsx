export interface ToDoFormProps {
  onAddTodo: (todo: string) => void;
  onRemoveTodo: () => void;
  onRemoveList: () => void;
}
