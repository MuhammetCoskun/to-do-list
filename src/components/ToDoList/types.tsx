export type ToDoListItemType = {
  id: string;
  text: string;
  createdAt: string;
  isDone: boolean;
  doneAt?: string;
};

export type ToDoListItemProps = {
  id: string;
  key?: string;
  text: string;
  onRemove?: (id: string) => void;
};

export type ToDoListProps = {
  list: ToDoListItemType[];
  onRemove?: (id: string) => void;
};
