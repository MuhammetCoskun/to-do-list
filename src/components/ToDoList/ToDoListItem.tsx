import React from "react";
import { ToDoListItemProps } from "./types";

const ToDoListItem: React.FC<ToDoListItemProps> = ({ id, text, onRemove }) => {
  const clickHandler = () => {
    onRemove?.(id);
  };

  return (
    <>
      <li onClick={clickHandler}>{text}</li>
    </>
  );
};

export default ToDoListItem;
