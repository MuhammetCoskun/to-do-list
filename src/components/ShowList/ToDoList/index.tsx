import React from "react";
import ToDoListItem from "./ToDoListItem";
import { ToDoListProps } from "./types";

const ToDoList: React.FC<ToDoListProps> = ({ list, onRemove }) => {
  return (
    <div className={"todo-tasks-container"}>
      {list.todos.map((item) => (
        <ToDoListItem
          key={item.id}
          id={item.id}
          text={item.text}
          onRemove={onRemove}
        />
      ))}
    </div>
  );
};

export default ToDoList;
