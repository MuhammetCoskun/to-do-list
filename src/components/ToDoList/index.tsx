import React from "react";
import ToDoListItem from "./ToDoListItem";
import { ToDoListProps } from "./types";

const ToDoList: React.FC<ToDoListProps> = ({ list, onRemove }) => {
  return (
    <ul>
      {list.map((item) => (
        <ToDoListItem
          key={item.id}
          id={item.id}
          text={item.text}
          onRemove={onRemove}
        />
      ))}
    </ul>
  );
};

export default ToDoList;
