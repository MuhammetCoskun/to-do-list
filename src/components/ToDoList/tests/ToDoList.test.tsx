import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ToDoList from "..";
import { ToDoListProps } from "../types";

const customRender = (props: ToDoListProps) => {
  render(<ToDoList {...props} />);
};
describe("ToDoList Component", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<ToDoList list={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("renders array elements", () => {
    customRender({
      list: [
        { id: "id", text: "Test text", createdAt: "2022", isDone: false },
        { id: "id2", text: "Test text2", createdAt: "2022", isDone: false },
      ],
    });
    const ulElement = screen.getByRole("list");
    expect(ulElement.childElementCount).toBe(2);
  });
});
