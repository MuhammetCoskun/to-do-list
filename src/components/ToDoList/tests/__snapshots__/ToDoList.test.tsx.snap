// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList Component renders correctly 1`] = `
<DocumentFragment>
  <ul />
</DocumentFragment>
`;
